### ISSUE TITLE
Session Access Rule [RBAC]

### ISSUE THREAD

**Issue by kontsevoy** [gravitational/teleport#1124]
## Proposal

Consider example:

```yaml
kind: role
version: v3
metadata:
  name: "developer"
spec:
  allow:
      rules:
          - resources: ["sessions"]
            verbs: ["play", "join", "list"]
  deny:
          - resources: ["sessions"]
            verbs: ["join"]
```

* `play` means a user can replay someone's session if he has the session ID via `tsh play`
* `join` means a user can join a session if he has the session ID.
* `list` means a user can see the list of sessions.

Notes:
* If a user does not have `list` verb he should not be able to see the list of sessions (active or not) in the UI.
* `play` without `list` is useful when a user is allowed to execute `tsh play <session-id>` but cannot see any sessions.
* `list` without `play` and without `join` is useful if a user wants to simply check if someone is using a particular machine.

The example above allows a user to do anything with the sessions except to join one. (because `deny` rules override `allow`).

### Web UI

* If a user does not have `sessions:list` the "sessions" tab should be hidden 
* If a user does not have `sessions:list` but types a proper URL, _authorization should be performed by the server_ and web UI should simply display the returned message.
* If a user does not have `sessions:list` but DOES have `sessions:play` he should be able to open a player for a particular session if someone sends him a URL.
* If a user does not have `sessions:join` permission, _authorization must be performed on the server_ and the web UI should simply display an error.

Basically all authorization should always be performed by the server. The web UI should simply display errors properly (and hide the session list, for convenience).

**Comment by kontsevoy** [comment]
Dupe of #1200
